### DISCUSSION TITLE
Import/edit mermaid code

### DISCUSSION THREAD

**Discussion by delorfin** [ted-marozzi/mermaid-flow-feedback#9] [Ideas]
Please consider adding the ability to import mermaid.js code that would then be rendered visually. Helpful e.g. when asking ChatGPT to provide a mermaid.js code and then go to Flow and continue with visual editing.

Another use case could be editing the current diagram in code instead of doing it visually — maybe to integrate a part of another diagram, to do bulk changes with regex in a text editor, etc.

**Comment by ted-marozzi** [comment]
I’m happy to say this feature already exists! Mermaid Flow supports both live editing of the diagram and the code in real time.

Here is an example diagram.
To import it into Mermaid Flow:
1) Open https://www.mermaidflow.app/flowchart
2) Open the “Export Panel”
3) Highlight all the text in the code editor
4) Paste the example diagram into the code editor
(Note only flowchart/graph diagrams are supported at this time.)

Mermaid flow will then render the diagram in the UI.

```
---
title: "Diagram Title"
---
flowchart
	this ----|"code"| can["be editied via the"]
	can --- and(("The UI"))
	and --- at("The Same time")
	%% Simply paste any mermaid flowchart you like in here
	%% and the UI will attempt to render it.

```

Let me know if this works as expected!

**Reply by delorfin** [reply]
Thank you, it does work indeed! Can't believe I haven't found it in the first place.

**Reply by ted-marozzi** [reply]
Don’t worry, you are not the first to ask me about this workflow and this feature! I would absolutely love if you could do a little screen recording showing me how you use it with ChatGPT, and even sharing it on social media. The more the word gets out about this tool the more motivated I am too work on it!

Also I pushed a small change out to make the import flow more obvious. :)
